### PR TITLE
Update asset dto class

### PR DESCRIPTION
**Pull request by stark-dev**
- add JSON serialization/deserialization
- change HashMap to ExtMap
- add direct access to ExtMap fields

Signed-off-by: Mauro Guerrera <mauroguerrera@eaton.com>

### CODE REVIEW

**Comment by stark-dev** [comment]
As discussed with @perrettecl

File: include/fty_asset_dto.h [hunk 1]
```diff
@@ -41,6 +40,9 @@ namespace fty
         Nonactive
     };
 
+    const std::string assetStatusToString(AssetStatus status);
+    AssetStatus stringToAssetStatus(const std::string & str);
```
Comment (etnKlendathu) [inline include/fty_asset_dto.h:44]: Are you sure to have those functions declared in header? Could be implemented as static in impl only.

File: include/fty_asset_dto.h [hunk 2]
```diff
@@ -127,7 +129,7 @@ namespace fty
     class Asset
     {
         public:
-        using HashMap = std::map<std::string, std::string>;
+        using ExtMap = std::map<std::string, std::pair<std::string,bool>>;
```
Comment (etnKlendathu) [inline include/fty_asset_dto.h:132]: On my teste, better to have struct in value part, not a pair with unsure parameters.

File: include/fty_asset_dto.h [hunk 3]
```diff
@@ -163,15 +168,17 @@ namespace fty
         std::string    m_parentId;
         // priority 1..5 (1 is most, 5 is least)
         int            m_priority      = 5;
-        // ext map storage (asset-specific values)
-        Asset::HashMap m_ext;
+        // ext map storage (asset-specific values with readonly attribute)
+        Asset::ExtMap m_ext;
```
Comment (etnKlendathu) [inline include/fty_asset_dto.h:172]: Asset prefix is redurant here.

File: include/fty_asset_dto.h [hunk 4]
```diff
@@ -163,15 +168,17 @@ namespace fty
...
     };
 
     void operator<<= (cxxtools::SerializationInfo & si, const Asset & asset);
     void operator>>= (const cxxtools::SerializationInfo & si, Asset & asset);
+    void operator<<= (std::string & json, const Asset & asset);
+    void operator>>= (const std::string & json, Asset & asset);
```
Comment (etnKlendathu) [inline include/fty_asset_dto.h:178]: <<= operator is ugly :) Maybe better to implement stream operatosr << and >> for std streams and use it more as convetional way.

File: include/fty_asset_dto.h (hunk 1 above)
Comment (stark-dev) [reply include/fty_asset_dto.h:44]: I need them in the asset server unfortunately, I may try to find another way though

File: include/fty_asset_dto.h (hunk 2 above)
Comment (stark-dev) [reply include/fty_asset_dto.h:132]: Keeping std::pair, as cxxtools is not able to serialize/deserialize structures on the fly. It's definitely less readable, but easier to handle as of now

File: include/fty_asset_dto.h (hunk 4 above)
Comment (stark-dev) [reply include/fty_asset_dto.h:178]: Converted to class methods

**Pull request merged**